hedera-node/src/main/java/com/hedera/services/contracts/execution/DirectCallsTxProcessor.java

```diff
@@ -0,0 +1,159 @@
+import org.hyperledger.besu.evm.gascalculator.GasCalculator;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import java.math.BigInteger;
+import java.time.Instant;
+import java.util.Map;
+import java.util.Optional;
+
+import static com.hedera.services.store.contracts.precompile.PrecompileMessage.State.COMPLETED_SUCCESS;
+import static com.hedera.services.store.contracts.precompile.PrecompileMessage.State.EXCEPTIONAL_HALT;
+import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.INSUFFICIENT_GAS;
+
+@Singleton
+public class DirectCallsTxProcessor {
+
+	private final GasCalculator gasCalculator;
+	private final LivePricesSource livePricesSource;
+	protected final GlobalDynamicProperties dynamicProperties;
+	private final HTSPrecompiledContract htsPrecompiledContract;
+	private final HederaMutableWorldState worldState;
+
+
+	@Inject
+	public DirectCallsTxProcessor(GasCalculator gasCalculator, LivePricesSource livePricesSource,
+								  GlobalDynamicProperties dynamicProperties, HTSPrecompiledContract htsPrecompiledContract, HederaMutableWorldState worldState) {
+		this.gasCalculator = gasCalculator;
+		this.livePricesSource = livePricesSource;
+		this.dynamicProperties = dynamicProperties;
+		this.htsPrecompiledContract = htsPrecompiledContract;
+		this.worldState = worldState;
+	}
+
+	public TransactionProcessingResult execute(
+			final Account sender,
+			final long gasLimit,
+			final long value,
+			final Bytes payload,
+			final Instant consensusTime,
+			final Address mirrorReceiver,
+			final BigInteger userOfferedGasPrice,
+			final long maxGasAllowanceInTinybars,
+			final Account relayer,
+			final Id tokenId,
+			final WorldLedgers ledgers) {
+
+		HederaWorldState.Updater updater = (HederaWorldState.Updater) worldState.updater();
+		long intrinsicGas = gasCalculator.transactionIntrinsicGasCost(Bytes.EMPTY, false);
+		final MutableAccount mutableSender = TxProcessorUtil.getMutableSender(updater, sender);
+		final var isEthTx = relayer != null;
+		final long gasPrice = TxProcessorUtil.gasPriceTinyBarsGiven(
+				livePricesSource, consensusTime, isEthTx,
+				HederaFunctionality.ContractCall);
+		final Wei gasCost = Wei.of(Math.multiplyExact(gasLimit, gasPrice));
+		final Wei upfrontCost = gasCost.add(value);
+
+		var allowanceCharged = Wei.ZERO;
+		MutableAccount mutableRelayer = null;
+		if (isEthTx) {
+			mutableRelayer = TxProcessorUtil.getMutableRelayer(updater, relayer);
+		}
+		if (intrinsicGas > gasLimit) {
+			throw new InvalidTransactionException(INSUFFICIENT_GAS);
+		}
+		if (!isEthTx) {
+			TxProcessorUtil.chargeGas(gasCost, upfrontCost, mutableSender);
+		} else {
+			allowanceCharged = TxProcessorUtil.chargeForEth(
+					userOfferedGasPrice, gasCost, maxGasAllowanceInTinybars, mutableSender,
+					mutableRelayer, Wei.ZERO, gasPrice, gasLimit, value);
+		}
+
+		PrecompileMessage message = PrecompileMessage.builder()
+				.setLedgers(ledgers)
+				.setSenderAddress(sender.canonicalAddress())
```

Comment:
⚠️ If the sender is an EOA (i.e., an auto-created account with an ECDSA key) then this will blow up any precompile that uses `senderAddress`. 

For example, I've pushed a branch with EET [here](https://github.com/hashgraph/hedera-services/blob/add-failing-eet/test-clients/src/main/java/com/hedera/services/bdd/suites/contract/hapi/ContractCallLocalSuite.java#L227) that pays for an ERC-20 `approve()` with an EOA; not only does the precompile incorrectly get `INVALID_ACCOUNT_ID`, the record includes,
```
sender_id {
  shardNum: -70672035
  realmNum: 8062295507940796406
  accountNum: 4428982008424832196
}
```